go/v1beta1/storage/elasticsearch.go

```diff
@@ -488,28 +564,83 @@ func (es *ElasticsearchStorage) DeleteOccurrence(ctx context.Context, pID, oID s
 
 // GetNote returns the note with project (pID) and note ID (nID)
 func (es *ElasticsearchStorage) GetNote(ctx context.Context, pID, nID string) (*pb.Note, error) {
-	return nil, nil
+	log := es.logger.Named("GetNote")
+
+	queryField := fmt.Sprintf("projects/%s/notes/%s", pID, nID)
+
+	var getDocumentBuffer bytes.Buffer
+	getDocumentBody := map[string]interface{}{
+		"query": map[string]interface{}{
+			"match": map[string]interface{}{
+				"name": queryField,
+			},
+		},
+	}
+	if err := json.NewEncoder(&getDocumentBuffer).Encode(getDocumentBody); err != nil {
+		return nil, createError(log, "error encoding elasticsearch get document body", err)
+	}
+
+	res, err := es.client.Search(
+		es.client.Search.WithContext(ctx),
+		es.client.Search.WithIndex(pID),
+		es.client.Search.WithBody(&getDocumentBuffer),
+	)
+	if err != nil {
+		return nil, createError(log, "error retrieving note from elasticsearch", err)
+	}
+	if res.StatusCode != http.StatusOK {
+		return nil, createError(log, "got unexpected status code from elasticsearch", nil, zap.Int("status", res.StatusCode))
+	}
+
+	log.Debug("elasticsearch response", zap.Any("response", res))
+
+	getDocumentResponse := &esSearchResponse{}
+	if err := json.NewDecoder(res.Body).Decode(&getDocumentResponse); err != nil {
+		log.Error("error decoding elasticsearch response body", zap.NamedError("error", err))
+		return nil, err
+	}
+
+	log.Debug("hit raw source", zap.Any("raw _source", getDocumentResponse.Hits.Hits[0].Source))
+
+	note := &pb.Note{}
+	jsonpb.Unmarshal(strings.NewReader(string(getDocumentResponse.Hits.Hits[0].Source)), note)
+
+	log.Debug("converted note", zap.Any("unmarshaled occurrence", note))
+
+	return note, nil
 }
 
 // ListNotes returns up to pageSize number of notes for this project (pID) beginning
 // at pageToken (or from start if pageToken is the empty string).
-func (es *ElasticsearchStorage) ListNotes(ctx context.Context, pID, filter, pageToken string, pageSize int32) ([]*pb.Note, string, error) {
+func (es *ElasticsearchStorage) ListNotes(ctx context.Context, projectID, filter, pageToken string, pageSize int32) ([]*pb.Note, string, error) {
 	log := es.logger.Named("ListNotes")
-	log.Debug("Project ID", zap.String("pID", pID))
+	log.Debug("Project ID", zap.String("projectID", projectID))
```

Comment:
I'd rather add this as a field to the current logger rather than log this on its own